File: react-performance/src/model/company.js
import { faker } from '@faker-js/faker';

export default class Company {
    name;
    catchPhrase;
    industry;
    phoneNumber;
    email;
    ipv4;

    constructor(name, catchPhrase, industry, phoneNumber, email, ipv4) {
        this.name = name;
        this.catchPhrase = catchPhrase;
        this.industry = industry;
        this.phoneNumber = phoneNumber;
        this.email = email;
        this.ipv4 = ipv4;
    }

    static generateFake() {
        const name = faker.company.companyName();
        const catchPhrase = faker.company.catchPhrase();
        const industry = faker.company.bsNoun();
        const phoneNumber = faker.phone.number();
        const email = faker.internet.email();
        const ipv4 = faker.internet.ip();
        return new Company(name, catchPhrase, industry, phoneNumber, email, ipv4);
    }

    /// Returns true if the text is contained in the object.
    /// The text is searched in the object's name, description, etc.
    /// The search is case insensitive.
    contains(text) {
        return this.name.toLowerCase().includes(text) ||
            this.catchPhrase.toLowerCase().includes(text) ||
            this.industry.toLowerCase().includes(text) ||
            this.phoneNumber.toLowerCase().includes(text) ||
            this.email.toLowerCase().includes(text) ||
            this.ipv4.toLowerCase().includes(text);
    }
}

File: rust-performance/src/model/company.rs
use fake::{Dummy, Fake};
use fake::faker::company::en::*;
use fake::faker::internet::en::*;
use fake::faker::phone_number::en::*;
use crate::extensions::search_text_filterable_trait::SearchTextFilterable;

#[derive(Debug, Dummy, PartialEq, Clone)]
pub struct Company {
    #[dummy(faker = "CompanyName()")]
    pub name: String,

    #[dummy(faker = "CatchPhase()")]
    pub catch_phrase: String,

    #[dummy(faker = "Industry()")]
    pub industry: String,

    #[dummy(faker = "PhoneNumber()")]
    pub phone_number: String,

    #[dummy(faker = "FreeEmail()")]
    pub email: String,

    #[dummy(faker = "IPv4()")]
    pub ipv4: String,
}

impl SearchTextFilterable for Company {
    fn contains(&self, text: &str) -> bool {
        self.name.to_lowercase().contains(text)
            || self.catch_phrase.to_lowercase().contains(text)
            || self.industry.to_lowercase().contains(text)
            || self.phone_number.to_lowercase().contains(text)
            || self.email.to_lowercase().contains(text)
            || self.ipv4.to_lowercase().contains(text)
    }
}